Admin spawnable version of the Death Ripley that actually inflicts damage and can rip arms off

Irafas opened the pull request:
:cl: 
add: Admin spawnable Death Ripley that comes with a version of the Death Clamp that actually inflicts damage and rips arms off.
/:cl:

The Death Ripley uses a KILL CLAMP rather than the normal clamp and overrides its action, though doesn't actually inflict any damage on the target. This fix simply inflicts that damage.

~~On disarm intent the KILL CLAMP is supposed to rip off the targets arms. As of writing this I'm not sure how to do this. I've been looking through trying to find how to do this, but any hints on where to look would be appreciated.~~
figured it out.

Irafas committed:
On Harm intent causes damage

Irafas committed:
Doesnt push people away when attacking them in harm intent

Irafas committed:
removed emtpy line

Irafas committed:
Revert "removed emtpy line"

This reverts commit 992bd05631f4d8d1b6a6aa851cd35137c4e94c73.

Irafas committed:
Now rips both arms off

Unless only the left arm is missing, in which case nothing happens and im not sure why. If the right arm is missing though the left arm is still ripped off.

Irafas commented:
For some reason if only the left arm is missing then the right arm won't be ripped off, but if the right arm is missing then the left one is. It has something to do with the order of them in the code, but i'm not sure why. I also need to have it add to the log when it rips off a persons arms.

SpaceManiac commented:
If the left arm is missing, `get_bodypart` will return null and the `affected.dismember` call will crash the proc.

Irafas committed:
Fixes issues with null values and such

added if statements that check if the affected limbs are null. Also added the attack to the logs.

Irafas commented:
Alright so this bug fix should be fixed now. The Kill Clamp now inflicts damage on the target and if the user is in disarm intent it now rips off the victims arms.

_**RIP'N'TEAR**_

RandomMarine commented:
Wasn't that always an intended joke?

Jordie0608 commented:
Yes the lack of damage is intentional.

Irafas commented:
Well I guess this was a waste of time. Still it made for some practice. If people want to keep the old joke clamps I'll close this PR.

Jordie0608 commented:
You could change this into another subtype of clamp that does actual damage if you wanted.

Irafas committed:
Makes a "proper" version of the death clamp

Creates a "proper" version of the death clamp that causes damage. Also added a "proper" version of the Death Ripley that includes said death clamp.

Irafas commented:
Alright so now there is a "proper" version of the death clamp that does actually inflict damage, and a "proper" version of the death ripley that comes equipped with it. The normal death ripley performs exactly like it did before and the new one is currently only spawnable by admins.

ShizCalev commented on code/game/mecha/equipment/tools/work_tools.dm:
```diff
@@ -108,11 +117,38 @@
+					affected.dismember(damtype)
+					play_sound = TRUE
+				if(play_sound)
+					playsound(get_turf(src), get_dismember_sound(), 80, 1)
```
TRUE at the end.

ShizCalev commented on code/game/mecha/equipment/tools/work_tools.dm:
```diff
@@ -108,11 +117,38 @@
+					playsound(get_turf(src), get_dismember_sound(), 80, 1)
+					target.visible_message("<span class='danger'>[chassis] rips [target]'s arms off.</span>", \
+								   "<span class='userdanger'>[chassis] rips [target]'s arms off.</span>")
+					add_logs(chassis.occupant, M, "attacked", "[name]", "(INTENT: [uppertext(chassis.occupant.a_intent)]) (DAMTYE: [uppertext(damtype)])")
```
be explicit about the action that happened instead of just "attacked"

Irafas committed:
ShizCalev's change requests

-Uses TRUE instead of 1.
-Used less stupid naming conventions and made the objects improper nouns (why did I even make them proper?)
-Gives a detailed log of which limbs were dismembered

Cyberboss commented on code/game/mecha/equipment/tools/work_tools.dm:
```diff
@@ -78,6 +78,14 @@
 	desc = "They won't know what clamped them!"
 	energy_drain = 0
+	var/real_clamp = FALSE
+	dam_force = 0
```
swap the lines of these two

Cyberboss commented on code/game/mecha/equipment/tools/work_tools.dm:
```diff
@@ -108,11 +116,41 @@
+					play_sound = TRUE
+					limbs_gone = "[limbs_gone], [affected]"
+				if(play_sound)
+					playsound(get_turf(src), get_dismember_sound(), 80, TRUE)
```
playsound(src, ...

Cyberboss commented on code/game/mecha/working/ripley.dm:
```diff
@@ -88,6 +88,17 @@
+	for(var/obj/item/mecha_parts/mecha_equipment/E in equipment)
+		E.detach()
+		qdel(E)
+	var/obj/item/mecha_parts/mecha_equipment/ME = new /obj/item/mecha_parts/mecha_equipment/hydraulic_clamp/kill/real
```
equipment.Cut() above this line.

Irafas committed:
Cyberboss's requested changes

-Moved variable declaration
-Playsound changed
-Added equipment.Cut() to initialization

AlexTheSysop42 commented:
Was the original not only adminspawn?
